File: expectanim/src/main/java/com/github/florent37/expectanim/core/scale/ScaleAnimExpectationSameHeightAs.java
package com.github.florent37.expectanim.core.scale;

import android.view.View;

/**
 * Created by florentchampigny on 20/02/2017.
 */
public class ScaleAnimExpectationSameHeightAs extends ScaleAnimExpectationViewDependant {

    public ScaleAnimExpectationSameHeightAs(View otherView, Integer gravityHorizontal, Integer gravityVertical) {
        super(otherView, gravityHorizontal, gravityVertical);
    }

    @Override
    public Float getCalculatedValueScaleX(View viewToMove) {
        if(keepRatio){
            return getCalculatedValueScaleY(viewToMove);
        }
        return null;
    }

    @Override
    public Float getCalculatedValueScaleY(View viewToMove) {
        final int viewToMoveHeight = viewToMove.getHeight();

        final float otherViewHeight = viewCalculator.finalHeightOfView(otherView);

        if (otherViewHeight == 0 || viewToMoveHeight == 0f) {
            return 0f;
        }
        return otherViewHeight / viewToMoveHeight;
    }
}

File: kotlinpleaseanimate/src/main/java/com/github/florent37/kotlin/pleaseanimate/core/scale/ScaleAnimExpectationSameHeightAs.kt
package com.github.florent37.kotlin.pleaseanimate.core.scale

import android.view.View

class ScaleAnimExpectationSameHeightAs(otherView: View, gravityHorizontal: Int?, gravityVertical: Int?) : ScaleAnimExpectationViewDependant(otherView, gravityHorizontal, gravityVertical) {

    override fun getCalculatedValueScaleX(viewToMove: View): Float? {
        return if (keepRatio) {
            getCalculatedValueScaleY(viewToMove)
        } else null
    }

    override fun getCalculatedValueScaleY(viewToMove: View): Float? {
        val viewToMoveHeight = viewToMove.height

        val otherViewHeight = viewCalculator!!.finalHeightOfView(otherView)

        return if (otherViewHeight == 0f || viewToMoveHeight.toFloat() == 0f) {
            0f
        } else otherViewHeight / viewToMoveHeight
    }
}
